nibabel/filebasedimages.py

```diff
@@ -239,16 +245,16 @@ def set_filename(self, filename: str):
         self.file_map = self.__class__.filespec_to_file_map(filename)
 
     @classmethod
-    def from_filename(klass, filename: FileSpec):
+    def from_filename(klass: type[ImgT], filename: FileSpec) -> ImgT:
         file_map = klass.filespec_to_file_map(filename)
         return klass.from_file_map(file_map)
 
     @classmethod
-    def from_file_map(klass, file_map: FileMap):
-        raise NotImplementedError
+    def from_file_map(klass: type[ImgT], file_map: FileMap) -> ImgT:
+        raise NotImplementedError  # pragma: no cover
 
     @classmethod
-    def filespec_to_file_map(klass, filespec: FileSpec):
+    def filespec_to_file_map(klass, filespec: FileSpec) -> FileMap:
```

Comment:
```suggestion
    def filespec_to_file_map(klass: type[ImgT], filespec: FileSpec) -> FileMap:
```
There are a few `classmethod`s here with this annotation and a few without. Seems to me like it would be better to keep it consistent and annotate all of them, if it doesn't break anything.

Reply:
Those `classmethods` return an image of a given type. Here the return type does not depend on the `klass` type, so we don't need to use a type variable. Using one type variable would fail in any case. If we want to be thorough, we could do:

```suggestion
    def filespec_to_file_map(klass: FileBasedImage, filespec: FileSpec) -> FileMap:
```

But this is just the default annotation, like saying `x: int = 1`. I'm -0 on it as clutter, but if you feel strongly I'm okay with adding it.